File: library/src/main/java/com/github/lany192/box/activity/ViewBindingActivity.java
package com.github.lany192.box.activity;

import android.os.Bundle;
import android.view.LayoutInflater;

import androidx.viewbinding.ViewBinding;

import java.lang.reflect.Method;
import java.lang.reflect.ParameterizedType;

public class ViewBindingActivity<T extends ViewBinding> extends EventBusActivity {
    protected T viewBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ParameterizedType type = (ParameterizedType) getClass().getGenericSuperclass();
        Class<?> cls = (Class<?>) type.getActualTypeArguments()[0];
        try {
            Method inflate = cls.getDeclaredMethod("inflate", LayoutInflater.class);
            viewBinding = (T) inflate.invoke(null, getLayoutInflater());
            setContentView(viewBinding.getRoot());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: library/src/main/java/com/github/lany192/box/activity/ViewBindingActivity.kt
package com.github.lany192.box.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewbinding.ViewBinding
import com.github.lany192.box.databinding.ToolbarDefaultBinding
import com.github.lany192.box.view.ActivityBinding

abstract class ViewBindingActivity<VB : ViewBinding> : AppCompatActivity() {
    private lateinit var _binding: ActivityBinding

    val binding get() = _binding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding =  ActivityBinding(layoutInflater, getToolbar(), getContentView())
        setContentView(binding.root)
        init(savedInstanceState)
    }

    fun getToolbar(): ViewBinding {
        return ToolbarDefaultBinding.inflate(layoutInflater)
    }

    abstract fun getContentView(): VB

    abstract fun init(savedInstanceState: Bundle?)
}